e2e/create/create_test.go

```diff
@@ -115,3 +123,128 @@ func TestCreatesSecureCluster(t *testing.T) {
 	}
 	steps.Run(t)
 }
+
+// TestCreateSecureClusterWithInvalidVersion tests cluster creation with invalid version and the cluster should fail.
+func TestCreateSecureClusterWithInvalidVersion(t *testing.T) {
+	// Test create a cluster with invalid version
+	// Check it went into ErrImagePull state and then marked CR into failed state
+	// tear it down
+
+	if parallel {
+		t.Parallel()
+	}
+	if testing.Short() {
+		t.Skip("skipping test in short mode.")
+	}
+
+	testLog := zapr.NewLogger(zaptest.NewLogger(t))
+
+	testcases := []struct {
+		imageVersion     string
+		cockroachVersion string
+	}{
+		{
+			nonExistentImage,
+			"",
+		},
+		{
+			validImage,
+			crdbVersion,
+		},
+	}
+
+	for _, testcase := range testcases {
+		steps := testutil.Steps{
+			{
+				Name: "creates 3-node secure cluster with invalid image",
+				Test: func(subT *testing.T) {
+					e := testenv.CreateActiveEnvForTest()
+					env := e.Start()
+
+					sb := testenv.NewDiffingSandbox(subT, env)
+					sb.StartManager(subT, controller.InitClusterReconcilerWithLogger(testLog))
+
+					builder := testutil.NewBuilder("crdb").WithNodeCount(3).WithTLS().
+						WithImage(testcase.imageVersion).
+						WithPVDataStore("1Gi", "standard" /* default storage class in KIND */)
+					if testcase.cockroachVersion != "" {
+						os.Setenv("RELATED_IMAGE_COCKROACH_v21_1_7", nonExistentImage)
```

Comment:
nit: make `"RELATED_IMAGE_COCKROACH_v21_1_7"` a var too, it's a little weird to have this hard-coded when everything else is a var